pynucastro/networks/rate_collection.py

```diff
@@ -801,11 +810,16 @@ def _evaluate_comp_NSE(self, u, rho, T, ye):
         # u_c is the coulomb correction term for NSE
         # Calculate the composition at NSE, equations found in appendix of Calder paper
         for nuc in self.unique_nuclei:
+            try:
```

Comment:
In the nucleus class I have defined the following:
```
        try:
            self.partition_function = _pcollection.get_partition_function(self.short_spec_name)
        except ValueError:
            self.partition_function = None```

Therefore, if the partition function of a nuclei is not define you will `get nuc.partition_function = None`. I suggest if
```
if not nuc.partition_function:
    pf = {lambda T: 1.0}
```

Reply:
I think it is good to point out to use `if nuc.partition_function` instead of `try/except` since raising exceptions is expensive and it is been triggered constantly over the loop. But I think it would be expensive to assign a lambda function continuously in a loop like this, especially when I just need a constant value?

Reply:
yes, you are right Zhi! with `pf=1.0` is also enough and cheaper.